### PR TITLE
TF >= 2.3 cleaning

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

The minimal TF version has recently been fixed to >=2.3, this PR remove all the <2.3 calls, mostly replacing experimental features by their stable ones.

### CODE REVIEW

File: src/transformers/trainer_tf.py [hunk 1]
```diff
@@ -93,11 +92,6 @@ def __init__(
             None,
         ),
     ):
-        assert parse(tf.__version__).release >= (2, 2, 0), (
```
Comment (patrickvonplaten) [inline src/transformers/trainer_tf.py:96]: Should that be updated to 2.3.0 in this case then?

**Review by patrickvonplaten — APPROVED** [review]

File: src/transformers/trainer_tf.py (hunk 1 above)
Comment (jplu) [reply src/transformers/trainer_tf.py:96]: No, because the min version for running the trainer was 2.2.0. So this assert becomes useless.

**Review by LysandreJik — APPROVED** [review]
LGTM

**Review by sgugger — APPROVED** [review]
LGTM, thanks for updating!

**Pull request merged**